src/js/edu-benefits/reducers/veteran/index.js

```diff
@@ -17,11 +30,21 @@ export default function veteran(state = blankVeteran, action) {
       return _.set(action.propertyPath, action.value, state);
     }
     case ENSURE_FIELDS_INITIALIZED: {
-      return action.fields.reduce((vet, field) => {
-        return _.isObject(vet[field]) ?
-          _.set([field, 'dirty'], true, vet) :
-          vet;
-      }, state);
+      const newState = Object.assign({}, state);
+
+      if (action.parentNode) {
+        action.fields.map((field) => {
+          Object.assign(newState[action.parentNode][0][field], dirtyAllFields(newState[action.parentNode][0][field]));
```

Comment:
I have a change for this in my page, too. One thing that seems weird about the way it is in HCA is that it's always updating the first record of the parentNode array. It seemed like this was causing problems with the GrowableTable I was using. 

In any case, I think this PR will be merged before mine, so I'll make sure whatever change I make works with this page and ping you to look at the change.